compiler/docgen.nim

```diff
@@ -431,36 +435,45 @@ proc getVarIdx(varnames: openArray[string], id: string): int =
       return i
   result = -1
 
-proc genComment(d: PDoc, n: PNode): PRstNode =
+func initItemFragment(rst: PRstNode): ItemFragment =
+  ItemFragment(isRst: true, rst: rst)
+
+func initItemFragment(str: string): ItemFragment =
+  ItemFragment(isRst: false, str: str)
+
+proc genComment(d: PDoc, n: PNode): Option[ItemFragment] =
   if n.comment.len > 0:
+    if optDocRaw in d.conf.globalOptions:
```

Comment:
Why check this recursively here and then wrap everything in an option when you can do the same at the callsite instead?

Reply:
My reasoning was that there is a few callsites for `genComment` so was safer to do it here in case I forgot somewhere